airflow/models/trigger.py

```diff
@@ -196,15 +196,16 @@ def assign_unassigned(cls, triggerer_id, capacity, session=None):
 
         # Find triggers who do NOT have an alive triggerer_id, and then assign
         # up to `capacity` of those to us.
-        trigger_ids_query = (
+        trigger_ids_query = with_row_locks(
             session.query(cls.id)
-            # notin_ doesn't find NULL rows
             .filter(or_(cls.triggerer_id.is_(None), cls.triggerer_id.notin_(alive_triggerer_ids)))
-            .limit(capacity)
-            .all()
-        )
-        session.query(cls).filter(cls.id.in_([i.id for i in trigger_ids_query])).update(
-            {cls.triggerer_id: triggerer_id},
-            synchronize_session=False,
-        )
+            .limit(capacity),
+            session,
+            skip_locked=True,
+        ).all()
+        if trigger_ids_query:
+            session.query(cls).filter(cls.id.in_([i.id for i in trigger_ids_query])).update(
```

Comment:
```suggestion
            session.query(cls).filter(cls.id.in_([i.id for i in trigger_ids_query])).update(
```

Would it be possible to transform `trigger_ids_query` to a subquery so this can be done in one db call instead of two?

Reply:
@uranusjr  As discussed update with limit is not supported, and adding subquery on same table is not supported in mysql